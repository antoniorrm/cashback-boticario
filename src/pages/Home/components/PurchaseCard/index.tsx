import React , { memo }from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { Card, CardContent, IconButton, Typography } from '@material-ui/core'
import { Delete, Edit } from '@material-ui/icons'

import { formatCurrecyBR } from '../../../../utils/formatCurrecyBR'

const useStyles = makeStyles(theme => ({
	root: {
		marginTop: '8px',
		marginBottom: '8px',
	},
	item: {
		margin: '8px',
		[theme.breakpoints.down('xs')]: {
			display: 'flex',
			alignItems: 'center',
			justifyContent: 'space-between',
		},
	},
	actionItem: {
		[theme.breakpoints.down('md')]: {
			display: 'flex',
			justifyContent: 'center',
		},
	},
	purchases: {
		[theme.breakpoints.up('sm')]: {
			display: 'flex',
		},
		alignItems: 'center',
		justifyContent: 'space-between',
	},
}))

type PurchaseCardProps = {
	codigo: string
	value: number
	valuePercent: number
	percent: number
	date: string
	status: string
	onEdit: () => void
	onDelete: () => void
}

function PurchaseCardComponent(props: PurchaseCardProps) {
	const classes = useStyles()

	const {
		codigo,
		value,
		valuePercent,
		percent,
		date,
		status,
		onEdit,
		onDelete,
	} = props

	return (
		<Card className={classes.root}>
			<CardContent className={classes.purchases}>
				<div className={classes.item}>
					<Typography color='textSecondary'>CÃ“DIGO</Typography>
					<Typography variant='body2' component='p'>
						{codigo}
					</Typography>
				</div>
				<div className={classes.item}>
					<Typography color='textSecondary'>VALOR</Typography>
					<Typography variant='body2' component='p'>
						{formatCurrecyBR(value)}
					</Typography>
				</div>
				<div className={classes.item}>
					<Typography color='textSecondary'>CASHBACK (%)</Typography>
					<Typography variant='body2' component='p'>
						{formatCurrecyBR(valuePercent)} ({percent}%)
					</Typography>
				</div>
				<div className={classes.item}>
					<Typography color='textSecondary'>DATA</Typography>
					<Typography variant='body2' component='p'>
						{date !== null && new Date(date).toLocaleDateString()}
					</Typography>
				</div>
				<div className={classes.item}>
					<Typography color='textSecondary'>STATUS</Typography>
					<Typography variant='body2' component='p'>
						{status}
					</Typography>
				</div>
				<div className={classes.actionItem}>
					<div>
						<IconButton aria-label='delete' color='primary' onClick={onEdit}>
							<Edit />
						</IconButton>
						<IconButton aria-label='delete' color='default' onClick={onDelete}>
							<Delete />
						</IconButton>
					</div>
				</div>
			</CardContent>
		</Card>
	)
}

export default memo(PurchaseCardComponent)