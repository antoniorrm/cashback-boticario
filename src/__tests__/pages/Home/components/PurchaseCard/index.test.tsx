/* eslint-disable no-undef */
import React from 'react'
import { render, screen } from '@testing-library/react'
import PurchaseCard from '../../../../../pages/Home/components/PurchaseCard'

const purchasePropsFake = {
	codigo: '1G5T',
	value: 100,
	valuePercent: 12,
	percent: 12,
	date: '2021-07-17T20:10:00.000Z',
	status: 'APROVADO',
	onEdit: () => {},
	onDelete: () => {},
}

describe('PurchaseCard render sucess', () => {
	it('renders component correct', () => {
		const { container } = render(
			<PurchaseCard
				codigo={purchasePropsFake.codigo}
				value={purchasePropsFake.value}
				valuePercent={purchasePropsFake.valuePercent}
				percent={purchasePropsFake.percent}
				date={purchasePropsFake.date}
				status={purchasePropsFake.status}
				onEdit={purchasePropsFake.onEdit}
				onDelete={purchasePropsFake.onDelete}
			/>
		)
		const testComponent = screen.getByText(/CÃ“DIGO/i)
		expect(container).toMatchSnapshot()
		expect(testComponent).toBeInTheDocument()
	})
})
