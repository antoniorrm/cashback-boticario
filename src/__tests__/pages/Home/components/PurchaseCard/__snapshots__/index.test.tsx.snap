// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseCard render sucess renders component correct 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-purchases-4"
    >
      <div
        class="makeStyles-item-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          CÓDIGO
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          1G5T
        </p>
      </div>
      <div
        class="makeStyles-item-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          VALOR
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          R$ 100,00
        </p>
      </div>
      <div
        class="makeStyles-item-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          CASHBACK (%)
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          R$ 12,00
           (
          12
          %)
        </p>
      </div>
      <div
        class="makeStyles-item-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          DATA
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          7/17/2021
        </p>
      </div>
      <div
        class="makeStyles-item-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          STATUS
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          APROVADO
        </p>
      </div>
      <div
        class="makeStyles-actionItem-3"
      >
        <div>
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="delete"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
